lmms_eval/models/vllm.py

```diff
@@ -0,0 +1,179 @@
+import asyncio
+import base64
+import json
+import os
+import time
+from copy import deepcopy
+from io import BytesIO
+from multiprocessing import cpu_count
+from typing import List, Tuple, Union
+
+import numpy as np
+from accelerate import Accelerator, DistributedType
+from decord import VideoReader, cpu
+from loguru import logger as eval_logger
+from openai import AsyncOpenAI, OpenAI
+from PIL import Image
+from tqdm import tqdm
+
+from lmms_eval.api.instance import Instance
+from lmms_eval.api.model import lmms
+from lmms_eval.api.registry import register_model
+
+NUM_SECONDS_TO_SLEEP = 5
+
+try:
+    import vllm
+except ImportError:
+    vllm = None
+
+
+@register_model("vllm")
+class VLLM(lmms):
```

Comment:
Since this processing is specific to VLMs, consider renaming to vllm_vlm

Reply:
I think this can remain as is since `vllm` supports both vision/language models and vision/language tasks.

This simply means supporting `vllm` through a single class.